File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task148.java

import java.util.EmptyStackException;

class Task148 {
    private static final int DEFAULT_CAPACITY = 10;
    private int[] stackArray;
    private int top;
    private int capacity;

    public Task148() {
        this(DEFAULT_CAPACITY);
    }

    public Task148(int capacity) {
        if (capacity <= 0) {
            throw new IllegalArgumentException("Capacity must be positive");
        }
        this.capacity = capacity;
        this.stackArray = new int[capacity];
        this.top = -1;
    }

    public void push(int value) {
        if (isFull()) {
            resize();
        }
        stackArray[++top] = value;
    }

    public int pop() {
        if (isEmpty()) {
            throw new EmptyStackException();
        }
        return stackArray[top--];
    }

    public int peek() {
        if (isEmpty()) {
            throw new EmptyStackException();
        }
        return stackArray[top];
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == capacity - 1;
    }

    public int size() {
        return top + 1;
    }

    private void resize() {
        int newCapacity = capacity * 2;
        int[] newArray = new int[newCapacity];
        System.arraycopy(stackArray, 0, newArray, 0, capacity);
        stackArray = newArray;
        capacity = newCapacity;
    }

    public static void main(String[] args) {
        System.out.println("=== Test Case 1: Basic Push and Pop ===");
        Task148 stack1 = new Task148();
        stack1.push(10);
        stack1.push(20);
        stack1.push(30);
        System.out.println("Peek: " + stack1.peek());
        System.out.println("Pop: " + stack1.pop());
        System.out.println("Pop: " + stack1.pop());
        System.out.println("Size: " + stack1.size());

        System.out.println("\\n=== Test Case 2: Empty Stack Check ===");
        Task148 stack2 = new Task148();
        System.out.println("Is Empty: " + stack2.isEmpty());
        stack2.push(5);
        System.out.println("Is Empty: " + stack2.isEmpty());
        stack2.pop();
        System.out.println("Is Empty after pop: " + stack2.isEmpty());

        System.out.println("\\n=== Test Case 3: Multiple Operations ===");
        Task148 stack3 = new Task148();
        for (int i = 1; i <= 5; i++) {
            stack3.push(i * 10);
        }
        System.out.println("Stack size: " + stack3.size());
        System.out.println("Peek: " + stack3.peek());
        while (!stack3.isEmpty()) {
            System.out.println("Pop: " + stack3.pop());
        }

        System.out.println("\\n=== Test Case 4: Stack Resize ===");
        Task148 stack4 = new Task148(2);
        stack4.push(1);
        stack4.push(2);
        stack4.push(3);
        stack4.push(4);
        System.out.println("Size after resize: " + stack4.size());
        System.out.println("Peek: " + stack4.peek());

        System.out.println("\\n=== Test Case 5: Error Handling ===");
        Task148 stack5 = new Task148();
        try {
            stack5.pop();
        } catch (EmptyStackException e) {
            System.out.println("Caught exception: Cannot pop from empty stack");
        }
        try {
            stack5.peek();
        } catch (EmptyStackException e) {
            System.out.println("Caught exception: Cannot peek empty stack");
        }
    }
}

File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task148.py

class Task148:
    def __init__(self, capacity=10):
        if capacity <= 0:
            raise ValueError("Capacity must be positive")
        self.capacity = capacity
        self.stack = []
        self.top = -1

    def push(self, value):
        if value is None:
            raise ValueError("Cannot push None value")
        self.stack.append(value)
        self.top += 1

    def pop(self):
        if self.is_empty():
            raise IndexError("Cannot pop from empty stack")
        self.top -= 1
        return self.stack.pop()

    def peek(self):
        if self.is_empty():
            raise IndexError("Cannot peek empty stack")
        return self.stack[self.top]

    def is_empty(self):
        return self.top == -1

    def size(self):
        return self.top + 1


def main():
    print("=== Test Case 1: Basic Push and Pop ===")
    stack1 = Task148()
    stack1.push(10)
    stack1.push(20)
    stack1.push(30)
    print(f"Peek: {stack1.peek()}")
    print(f"Pop: {stack1.pop()}")
    print(f"Pop: {stack1.pop()}")
    print(f"Size: {stack1.size()}")

    print("\\n=== Test Case 2: Empty Stack Check ===")
    stack2 = Task148()
    print(f"Is Empty: {stack2.is_empty()}")
    stack2.push(5)
    print(f"Is Empty: {stack2.is_empty()}")
    stack2.pop()
    print(f"Is Empty after pop: {stack2.is_empty()}")

    print("\\n=== Test Case 3: Multiple Operations ===")
    stack3 = Task148()
    for i in range(1, 6):
        stack3.push(i * 10)
    print(f"Stack size: {stack3.size()}")
    print(f"Peek: {stack3.peek()}")
    while not stack3.is_empty():
        print(f"Pop: {stack3.pop()}")

    print("\\n=== Test Case 4: Large Stack ===")
    stack4 = Task148()
    for i in range(100):
        stack4.push(i)
    print(f"Size: {stack4.size()}")
    print(f"Peek: {stack4.peek()}")

    print("\\n=== Test Case 5: Error Handling ===")
    stack5 = Task148()
    try:
        stack5.pop()
    except IndexError as e:
        print(f"Caught exception: {e}")
    try:
        stack5.peek()
    except IndexError as e:
        print(f"Caught exception: {e}")


if __name__ == "__main__":
    main()
